repr_test.go

```diff
@@ -97,3 +97,54 @@ func TestMarshalAminoJSON(t *testing.T) {
 	assert.Equal(t, f, f2)
 	assert.Equal(t, f.a, f2.a) // In case the above doesn't check private fields?
 }
+
+type Bar struct {
+	a string
+	b int
+	c []*Bar
+	D string // exposed
+}
+
+func (b Bar) MarshalAminoJSON() ([]pair, error) { // nolint: golint
+	return []pair{
```

Comment:
Curious, why does this return a list of pairs and not bytes? What is the typical implementation of `MarshalAminoJSON` look like?

Reply:
Yeah, it can return whatever it wants. I just copied the `MarshalAmino` tests, which returned pairs.

In the SDK we'll want to return []byte, like here: https://github.com/cosmos/cosmos-sdk/pull/7276/files#diff-1f6fdece2a5550f27ed0b77247d58f8bR60-R65